Eeprom file written correctly with avrdude 7.1 generates error with 7.2

Overview                                                                                                                                 
                                                                                                                                         
An eeprom file, which is correctly written to an atmega328p by avrdude 6.3 7.0 and 7.1, generates a verification error when written with avrdude 7.2. Looking at the eeprom in -t mode shows that the eeprom has not been correctly written.                                                                                                   
                                                                                                                                         
Example error                                                                                                                            
                                                                                                                                         
The original eeprom file is relatively large but a similar error                                                                      
can be reproduced with this small hex file.                                                                                              
                                                                                                                                         
With 7.2 this hex file is incorrectly written, but with 7.1 no error                                                                     
```                                                                                                                                         
:02000200AABB97
:00000001FF
```                                                                                                                                         
I tried a few more examples                                                                                                              
                                                                                                                                         
Add a line and it is correctly written with 7.2 and 7.1                                                                                  
```                                                                                                                                         
:02000000DDCC55
:02000200AABB97
:00000001FF
```                                                                                                                                         
But this file is also correctly written with 7.2 and 7.1                                                                                 
```                                                                                                                                         
:02000400AABB95
:00000001FF
```                                                                                                                                         
And finally this is the line from the original file which caused the                                                                     
error with 7.2 but not 7.1 7.0 and 6.3                                                                                                   
```                                                                                                                                         
:10003400FFFFF00AC4019000F1095C00B008EB0A6C
:00000001FF
```
Is is possible to confirm this behaviour (outside of my setup)?

Kind regards and thanks,
Tristan

HEX FILE one.eep.hex                                                                                                                     
                                                                                                                                         
:02000200AABB97                                                                                                                          
:00000001FF                                                                                                                              
                                                                                                                                         
OUTPUT FROM 7.2                                                                                                                          
                                                                                                                                         
/usr/local/bin/avrdude  -c usbtiny -p m328p -e -U eeprom:w:one.eep.hex:i                                                                 
avrdude: AVR device initialized and ready to accept instructions                                                                         
avrdude: device signature = 0x1e950f (probably m328p)                                                                                    
avrdude: erasing chip                                                                                                                    
                                                                                                                                         
avrdude: processing -U eeprom:w:one.eep.hex:i                                                                                            
avrdude: reading input file one.eep.hex for eeprom                                                                                       
         with 2 bytes in 1 section within [2, 3]                                                                                         
         using 1 page and 2 pad bytes                                                                                                    
avrdude: writing 2 bytes eeprom ...                                                                                                      
Writing | ################################################## | 100% 0.03s                                                                
avrdude: 2 bytes of eeprom written                                                                                                       
avrdude: verifying eeprom memory against one.eep.hex                                                                                     
Reading | ################################################## | 100% 0.00s                                                                
avrdude warning: verification mismatch                                                                                                   
        device 0xff != input 0xaa at addr 0x0002 (error)                                                                                 
avrdude error: verification mismatch                                                                                                     
                                                                                                                                         
avrdude done.  Thank you.                                                                                                                
                                                                                                                                         
make: *** [one] Error 1                                                                                                                  
                                                                                                                                         
OUTPUT FROM 7.1                                                                                                                          
                                                                                                                                         
/Users/tw/Downloads/avrdude-7.1/build_darwin/src/avrdude  -c usbtiny -p m328p -e -U eeprom:w:one.eep.hex:i                               
                                                                                                                                         
avrdude: AVR device initialized and ready to accept instructions                                                                         
avrdude: device signature = 0x1e950f (probably m328p)                                                                                    
avrdude: erasing chip                                                                                                                    
avrdude: reading input file one.eep.hex for eeprom                                                                                       
         with 2 bytes in 1 section within [2, 3]                                                                                         
         using 1 page and 2 pad bytes                                                                                                    
avrdude: writing 2 bytes eeprom ...                                                                                                      
                                                                                                                                         
Writing | ################################################## | 100% 0.04s                                                                
                                                                                                                                         
avrdude: 2 bytes of eeprom written                                                                                                       
avrdude: verifying eeprom memory against one.eep.hex                                                                                     
                                                                                                                                         
Reading | ################################################## | 100% 0.00s                                                                
                                                                                                                                         
avrdude: 2 bytes of eeprom verified                                                                                                      
                                                                                                                                         
avrdude done.  Thank you.

Thank you for addressing this issue.  I believe I have built PR #1511 I have included the default help output so I would grateful if you could check that I am running the required version. 

Sadly, it does not work for my (a generic 2313A based) usbtiny.  Both the example hex file and the actual hex (that highlighted the issue to me give) verification errors. 

Usage: avrdude [options]                                                                               
Options:                                                                                               
  -p <partno>            Specify AVR device; -p ? lists all known parts                                
  -p <wildcard>/<flags>  Run developer options for matched AVR devices,                                
                         e.g., -p ATmega328P/s or /S for part definition                               
  -b <baudrate>          Override RS-232 baud rate                                                     
  -B <bitclock>          Specify bit clock period (us)                                                 
  -C <config-file>       Specify location of configuration file                                        
  -c <programmer>        Specify programmer; -c ? and -c ?type list all                                
  -c <wildcard>/<flags>  Run developer options for matched programmers,                                
                         e.g., -c 'ur*'/s for programmer info/definition                               
  -A                     Disable trailing-0xff removal for file/AVR read                               
  -D                     Disable auto erase for flash memory; implies -A                               
  -i <delay>             ISP Clock Delay [in microseconds]                                             
  -P <port>              Connection; -P ?s or -P ?sa lists serial ones                                 
  -F                     Override invalid signature or initial checks                                  
  -e                     Perform a chip erase                                                          
  -O                     Perform RC oscillator calibration (see AVR053)                                
  -t                     Run an interactive terminal when it is its turn                               
  -T <terminal cmd line> Run terminal line when it is its turn                                         
  -U <memtype>:r|w|v:<filename>[:format]                                                               
                         Carry out memory operation when it is its turn                                
                         Multiple -t, -T and -U options can be specified                               
  -n                     Do not write to the device whilst processing -U                               
  -V                     Do not automatically verify during -U                                         
  -E <exitsp>[,<exitsp>] List programmer exit specifications                                           
  -x <extended_param>    Pass <extended_param> to programmer, see -xhelp                               
  -v                     Verbose output; -v -v for more                                                
  -q                     Quell progress output; -q -q for less                                         
  -l logfile             Use logfile rather than stderr for diagnostics                                
  -?                     Display this usage                                                            
                                                                                                       
avrdude version 7.2-20231006 (c8b4e915), https://github.com/avrdudes/avrdude       

:02000200AABB97                                                                                        
:00000001FF           

debuging...                                                                                            
/Users/tw/avrdude/build_darwin/src/avrdude  -c usbtiny  -p m328p -e -U eeprom:w:one.eep.hex:i          
avrdude: AVR device initialized and ready to accept instructions                                       
avrdude: device signature = 0x1e950f (probably m328p)                                                  
avrdude: erasing chip                                                                                  
                                                                                                       
avrdude: processing -U eeprom:w:one.eep.hex:i                                                          
avrdude: reading input file one.eep.hex for eeprom                                                     
         with 2 bytes in 1 section within [2, 3]                                                       
         using 1 page and 2 pad bytes                                                                  
avrdude: writing 2 bytes eeprom ...                                                                    
Writing | ################################################## | 100% 0.03s                              
avrdude: 2 bytes of eeprom written                                                                     
avrdude: verifying eeprom memory against one.eep.hex                                                   
Reading | ################################################## | 100% 0.00s                              
avrdude warning: verification mismatch                                                                 
        device 0xff != input 0xaa at addr 0x0002 (error)                                               
avrdude error: verification mismatch                                                                   
                                                                                                       
avrdude done.  Thank you.                                                                              
                                                                                                       
make: *** [one] Error 1                                                                                
                          
:10003400FFFFF00AC4019000F1095C00B008EB0A6C                                                            
:0A00440006034800BB3F010048001E                                                                        
:06005C000200840670069C                                                                                
:10006600733B680000000000FF08AF08AF080000FF                                                            
:100076000A00A300B100780093007E0200006B0224                                                            
:0A008600A93CC83CB83CE33C19005B                                                                        
:00000001FF 

Uploading Hexfiles to Arduino uno                                                                      
/Users/tw/avrdude/build_darwin/src/avrdude  -c usbtiny  -p m328p -e -U flash:w:uno.hex:i -U eeprom:w:u\
no.eep.hex:i                                                                                           
avrdude: AVR device initialized and ready to accept instructions                                       
avrdude: device signature = 0x1e950f (probably m328p)                                                  
avrdude: erasing chip                                                                                  
                                                                                                       
avrdude: processing -U flash:w:uno.hex:i                                                               
avrdude: reading input file uno.hex for flash                                                          
         with 9522 bytes in 26 sections within [0, 0x7f81]                                             
         using 76 pages and 206 pad bytes                                                              
avrdude: writing 9522 bytes flash ...                                                                  
Writing | ################################################## | 100% 7.36s                              
avrdude: 9522 bytes of flash written                                                                   
avrdude: verifying flash memory against uno.hex                                                        
Reading | ################################################## | 100% 5.42s                              
avrdude: 9522 bytes of flash verified                                                                  
                                                                                                       
avrdude: processing -U eeprom:w:uno.eep.hex:i                                                          
avrdude: reading input file uno.eep.hex for eeprom                                                     
         with 74 bytes in 3 sections within [0x34, 0x8f]                                               
         using 20 pages and 6 pad bytes                                                                
avrdude: writing 74 bytes eeprom ...                                                                   
Writing | ################################################## | 100% 0.21s                              
avrdude: 74 bytes of eeprom written                                                                    
avrdude: verifying eeprom memory against uno.eep.hex                                                   
Reading | ################################################## | 100% 0.03s                              
avrdude warning: verification mismatch                                                                 
        device 0xff != input 0xf0 at addr 0x0036 (error)                                               
avrdude error: verification mismatch                                                                   
                                                                                                       
avrdude done.  Thank you.                                                                              
                                                                                                       
make: *** [uno] Error 1

--- a/src/usbtiny.c
+++ b/src/usbtiny.c
@@ -721,9 +721,9 @@ static int usbtiny_paged_write(const PROGRAMMER *pgm, const AVRPART *p, const AV
 
   delay = 0;
   if (! m->paged) {
-    unsigned int poll_value;
-    // Does this chip not support paged writes?
-    poll_value = (m->readback[1] << 8) | m->readback[0];
+    unsigned int poll_value = (m->readback[1] << 8) | m->readback[0];
+    if(!poll_value)
+      poll_value = 0xffff;
     if (usb_control(pgm, USBTINY_POLL_BYTES, poll_value, 0 ) < 0)
       return -1;
     delay = m->max_write_delay;

